feature-staking-impl/src/main/java/jp/co/soramitsu/staking/impl/presentation/staking/main/StakingViewModel.kt

```diff
@@ -110,12 +125,61 @@ class StakingViewModel @Inject constructor(
             it.alerts()
         }.distinctUntilChanged().shareIn(stakingStateScope, started = SharingStarted.Eagerly, replay = 1)
 
+    private val defaultNetworkInfoStates = mapOf(
+        StakingType.POOL to StakingAssetInfoViewState.StakingPool.default(resourceManager),
+        StakingType.RELAYCHAIN to StakingAssetInfoViewState.RelayChain.default(resourceManager),
+        StakingType.PARACHAIN to StakingAssetInfoViewState.Parachain.default(resourceManager)
+    )
+
+    private val stakingViewState1: SharedFlow<StakingViewState1?> = scenarioViewModelFlow
+        .flatMapLatest {
+            it.getStakingViewStateFlow1()
+        }.distinctUntilChanged().shareIn(stakingStateScope, started = SharingStarted.Eagerly, replay = 1)
+
+    inline fun <reified T : StakingAssetInfoViewState> Map<StakingType, StakingAssetInfoViewState>.get(type: StakingType): T = get(type) as T
+
+    private val networkInfoState: Flow<StakingAssetInfoViewState?> = networkInfo.map { networkInfoState ->
+        val selection = stakingSharedState.selectionItem.first()
+        if (selection.type != StakingType.POOL) return@map null
+        if (networkInfoState is LoadingState.Loaded) {
+            when (val state = networkInfoState.data) {
+                is StakingNetworkInfoModel.Parachain -> defaultNetworkInfoStates.get<StakingAssetInfoViewState.Parachain>(StakingType.PARACHAIN).update(state)
+                is StakingNetworkInfoModel.Pool -> defaultNetworkInfoStates.get<StakingAssetInfoViewState.StakingPool>(StakingType.POOL).update(state)
```

Comment:
are other variants possible with previous condition "143   if (selection.type != StakingType.POOL) return@map null" ?

Reply:
if (selection.type != StakingType.POOL) this condition is just a stub. The idea is to show everything regarding pools in compose, but other types of staking - in xml. So I avoid here any other staking types. But I added it here because I make components that will be ready for all staking types.